docs/map.xsd

```diff
@@ -204,41 +365,92 @@
 <xs:attributeGroup name="property">
   <xs:attribute name="name" type="nameT" use="required"/>
   <xs:attribute name="value" type="xs:string" use="required"/>
+  <xs:attribute name="type" type="propertyT" use="optional" default="string"/>
 </xs:attributeGroup>
 
 <xs:attributeGroup name="map">
   <xs:attribute name="version" type="xs:string" use="required"/>
   <xs:attribute name="orientation" type="orientationT" use="required"/>
-  <xs:attribute name="renderorder" type="renderorderT"/>
-  <xs:attribute name="width" type="xs:nonNegativeInteger" use="required"/>
-  <xs:attribute name="height" type="xs:nonNegativeInteger" use="required"/>
-  <xs:attribute name="tilewidth" type="xs:nonNegativeInteger" use="required"/>
-  <xs:attribute name="tileheight" type="xs:nonNegativeInteger" use="required"/>
+  <xs:attribute name="renderorder" type="renderorderT" default="right-down"/>
+  <xs:attribute name="width" type="pixelT" use="required"/>
+  <xs:attribute name="height" type="pixelT" use="required"/>
+  <xs:attribute name="tilewidth" type="pixelT" use="required"/>
+  <xs:attribute name="tileheight" type="pixelT" use="required"/>
+  <xs:attribute name="hexsidelength" type="pixelT" use="optional"/>
+  <xs:attribute name="staggeraxis" type="staggeraxisT" use="optional"/>
+  <xs:attribute name="staggerindex" type="staggerindexT" use="optional"/>
+  <xs:attribute name="backgroundcolor" type="hexColourAlphaT" use="optional"/>
+  <xs:attribute name="nextobjectid" type="xs:nonNegativeInteger" use="required"/>
 </xs:attributeGroup>
 
 <xs:attributeGroup name="tileset">
   <xs:attribute name="name" type="nameT"/>
   <xs:attribute name="firstgid" type="xs:positiveInteger" use="required"/>
   <xs:attribute name="source" type="xs:anyURI"/>
-  <xs:attribute name="tilewidth" type="xs:nonNegativeInteger"/>
-  <xs:attribute name="tileheight" type="xs:nonNegativeInteger"/>
-  <xs:attribute name="spacing" type="xs:nonNegativeInteger"/>
+  <xs:attribute name="tilewidth" type="pixelT"/>
+  <xs:attribute name="tileheight" type="pixelT"/>
+  <xs:attribute name="spacing" type="pixelT"/>
+  <xs:attribute name="margin" type="pixelT"/>
+  <xs:attribute name="tilecount" type="xs:nonNegativeInteger"/>
+  <xs:attribute name="columns" type="xs:nonNegativeInteger"/>
 </xs:attributeGroup>
 
-<xs:attributeGroup name="image.tileset">
+  <xs:attributeGroup name="imagelayer">
+    <xs:attribute name="name" type="nameT"/>
+    <xs:attribute name="offsetx" type="pixelT" default="0" use="optional"/>
+    <xs:attribute name="offsety" type="pixelT" default="0" use="optional"/>
```

Comment:
The offset can actually be negative, so the non-negative `pixelT` would be the wrong type here. Maybe just use `xs:integer`?